### PR TITLE
added consistent dest_surface description in transform.rst

### PR DESCRIPTION
**Pull request by VantaTree**

### CODE REVIEW

**Comment by Starbuck5** [comment]
Depth and format are not interchangeable.

You could have 2 32 bit surfaces, one of which is RGBA and one of which is BGRA.

**Comment by VantaTree** [comment]
Results from testing each surface
```
S: same
A: any
SF: same as first
~: not tested

|   func name    |depth|format|
|scale           |  S  |  S  |
|scale_by        |  S  |  S  |
|smoothscale     |  S  |  S  |
|smoothscale_by  |  S  |  S  |
|laplacian       |  S  |  A  |
|box_blur        |  S  |  S  |
|gaussian_blur   |  S  |  S  |
|invert          |  S  |  A  |
|grayscale       |  S  |  A  |
|average_surfaces|  SF |  A  |
```

**Review by MyreMylar — APPROVED** [review]
LGTM 👍 

The description is clear, and probably exposes a few places where we should tidy up how the functions work when passing destinations surface with non-matching pixel surfaces. I personally thing that we should be checking that the destination surface has a matching pixel format as I don't really want to add format conversion on top of what each function is doing - especially as passing a destination surface is supposed to be 'speeding up' the process.

File: docs/reST/ref/transform.rst [hunk 1]
```diff
@@ -42,11 +42,9 @@ Instead, always begin with the original image and scale to the desired size.)
...
 
-   An optional destination surface can be used, rather than have it create a
-   new one. This is quicker if you want to repeatedly scale something. However
-   the destination must be the same size as the size (width, height) passed in. Also
-   the destination surface must be the same format.
-
+   An optional destination surface can be passed which is faster than creating a new
+   Surface. This destination surface must be the same as the size (width, height) passed
+   in, and the same depth and format as the source Surface.
    .. ## pygame.transform.scale ##
```
Comment (Starbuck5) [inline docs/reST/ref/transform.rst:48]: There needs to be a space in between these two elements.

I tested it, without the space the paragraph doesn't generate at all:
Capture

**Comment by Starbuck5** [comment]
Needs space added (as per comment), needs merge conflict resolution. 

Once those two small things are sorted this looks good for merge!

I'd also like a list of the functions that accept differing formats and then fail. Those should be changed. (Probably to just error instead)

**Review by Starbuck5 — APPROVED** [review]
Thanks!

**Pull request merged**